src/components/Personalization/createApplyPropositions.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import composePersonalizationResultingObject from "./utils/composePersonalizationResultingObject";
+import isNonEmptyArray from "../../utils/isNonEmptyArray";
+import isEmptyObject from "../../utils/isEmptyObject";
+import isObject from "../../utils/isObject";
+
+export const EMPTY_PROPOSITIONS = { propositions: [] };
+export const HTML_CONTENT_ITEM = "html-content-item";
+export const DEFAULT_METADATA = {
+  selector: "head",
+  type: "appendHtml"
+};
+
+export default ({ executeDecisions, showContainers }) => {
+  const updatePropositions = ({ proposition, metadataForScope }) => {
+    proposition.items.forEach(item => {
+      if (item.schema.includes(HTML_CONTENT_ITEM)) {
+        item.data.selector =
+          metadataForScope.selector || DEFAULT_METADATA.selector;
+        item.data.type = metadataForScope.actionType || DEFAULT_METADATA.type;
```

Comment:
Should we be making changes to the propositions the user is sending in? This could leak across calls to applyPropositions.

Reply:
On line 40 I copy the proposition before modifying it:
`completeProposition = { ...proposition }`
But maybe to make it more clear I can use `Array.from(propositions)` in the first line of applyPropositions?

Reply:
No, I don't think you need to change anything. I just didn't look close enough.